Add `from_timestamp_secs` method to `DateTime`

jasonaowen opened the pull request:
Add a new single-argument function for creating a `DateTime` from a Unix timestamp, to match `from_timestamp_micros`, `from_timestamp_millis`, and `from_timestamp_nanos`.

Resolves #1716 Suggestion: add DateTime::from_timestamp_secs function

djc commented on src/datetime/mod.rs:
```diff
@@ -755,6 +755,61 @@ impl DateTime<Utc> {
+    #[inline]
+    #[must_use]
+    pub const fn from_timestamp_secs(secs: i64) -> Option<Self> {
+        Self::from_timestamp(secs, 0)
```
Yes, would be nice!

djc commented on src/datetime/mod.rs:
```diff
@@ -755,6 +755,61 @@ impl DateTime<Utc> {
+    ///
+    #[inline]
+    #[must_use]
+    pub const fn from_timestamp_secs(secs: i64) -> Option<Self> {
```
I'd like to move this before `from_timestamp()`, because it depends on it.

jasonaowen replied:
Currently `from_timestamp` is the first method in the `impl DateTime<Utc>`, and the other `from_timestamp_*` methods follow it. Should I move the others, as well?

djc replied:
Just put this new one in the right place, thanks.

jasonaowen commented:
I slimmed down the tests, refactored the existing code to use `from_timestamp_secs` and `from_timestamp_millis`, and rebased onto `main`. I think this is ready to re-review!

djc commented:
> I slimmed down the tests, refactored the existing code to use `from_timestamp_secs` and `from_timestamp_millis`, and rebased onto `main`. I think this is ready to re-review!

Please squash your commits into a single one.

jasonaowen commented:
Relocated and squashed!
